compiler/plugins/input/Torch/InputConversion/ConvertTorchUnstructuredToLinalgExt.cpp

```diff
@@ -163,6 +171,349 @@ struct FftRfftOpConversion
   }
 };
 
+struct FlexAttentionOpConversion
+    : public OpRewritePattern<torch::Torch::AtenFlexAttentionOp> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(torch::Torch::AtenFlexAttentionOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    MLIRContext *ctx = getContext();
+    // Extract operands (new order: query, key, value, 8 block_mask tensors,
+    // scale, return_lse)
+    Value query = op.getQuery();
+    Value key = op.getKey();
+    Value value = op.getValue();
+    Value scaleValue = op.getScale();
+    Value returnLseValue = op.getReturnLse();
+
+    // Get tensor types
+    auto queryType = cast<torch::Torch::ValueTensorType>(query.getType());
+    auto keyType = cast<torch::Torch::ValueTensorType>(key.getType());
+    auto valueType = cast<torch::Torch::ValueTensorType>(value.getType());
+
+    // TODO: See if this check is necessary (Op assertations)
+    if (!queryType.hasSizes() || !keyType.hasSizes() || !valueType.hasSizes()) {
+      return rewriter.notifyMatchFailure(op,
+                                         "expected tensors with known sizes");
+    }
+
+    ArrayRef<int64_t> queryShape = queryType.getSizes();
+    ArrayRef<int64_t> valueShape = valueType.getSizes();
+
+    // TODO: See if this check is necessary (Op assertations)
+    // Query shape: [B, H, M, E]
+    if (queryShape.size() != 4) {
+      return rewriter.notifyMatchFailure(op, "expected 4D query tensor");
+    }
+
+    int64_t batch = queryShape[0];
+    int64_t numHeads = queryShape[1];
+    int64_t seqLenQ = queryShape[2];
+    int64_t headDim = queryShape[3];
+    int64_t seqLenKV = keyType.getSizes()[2];
+    int64_t valueDim = valueShape[3];
+
+    if (headDim == torch::Torch::kUnknownSize) {
+      return rewriter.notifyMatchFailure(op, "NYI: dynamic head dimension");
+    }
+
+    // Get element type
+    Type elementType = queryType.getOptionalDtype();
+    if (!elementType || !isa<FloatType>(elementType)) {
+      return rewriter.notifyMatchFailure(op, "expected float element type");
+    }
+    FloatType floatType = cast<FloatType>(elementType);
+
+    // Handle scale parameter
+    // Default scale: 1.0 / sqrt(head_dim)
+    double scaleVal;
+    if (!matchPattern(scaleValue,
+                      torch::Torch::m_TorchConstantFloat(&scaleVal)))
+      // Fallback to default scale
+      scaleVal = 1.0 / std::sqrt(static_cast<double>(headDim));
```

Comment:
IREE style nit: wrap braces. Please check other code as well.

note: you also need braces even you are following LLVM style: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements